example_def.json_CFE-3612

```diff
@@ -0,0 +1,4 @@
+"vars": {
+                "packages_platform_default_alternate_aix": "yum",
```

Comment:
Humm, I am not sure about this platform specific setting.

My initial thought is to expose only a single tunable for the default platform. However, that would mean that you could not use just a simple `def.json` if you had mixed platforms, since the setting would take effect for all platforms. In order to have a different setting for a different platform, you would need to use multiple augments support, e.g. `"augments": [ "$(sys.flavor).json" ]`. Then you could for example add an `aix.json` where `packages_platform_default` is set to `yum`.

On the other hand using a separate key for each platform like this makes it very simple to set different defaults for different platforms (I am imagining ahead where there might also be a `packages_platform_default_alternate_ubuntu` or something). However, it's also very plausible that you might only want to do that for a subset of infrastructure, or for a certain version of a platform or higher, at which point you would need to head down the multiple augments path or direct policy edits.

Reply:
Why not make this a `data` variable with the platforms as keys?

Reply:
Ah, that's kind of clever.

```
{
  "vars": {
    "packages_platform_default": {
        "$(sys.flavor)": "flavor_default",
        "ubuntu_20": "apt_get",
        "aix?": "yum"
      }
  }
}
```

Reply:
anything i can do / test to give this a push forward?